### PR TITLE
CRM-15762: Modify releaser script to update versions.json

### PR DESCRIPTION
**Pull request by twomice**
I hope this makes sense.

### CODE REVIEW

**Comment by totten** [comment]
jenkins, add to white list

File: tools/scripts/releaser/releaser [hunk 1]
```diff
@@ -141,27 +186,82 @@ function do_distmaker() {
...
+
+    mv *.tar.gz *.tgz *.zip *MD5SUMS* $build_dest
+
+  popd
+}
+
+#################################################
+## Publish version data to versions.json
+function do_publish_versions() {
+  # FIXME: Is this racy when doing concurrent security releases?
```
Comment (colemanw) [inline tools/scripts/releaser/releaser]: Not anymore :)

File: tools/scripts/releaser/releaser [hunk 2]
```diff
@@ -141,27 +186,80 @@ function do_distmaker() {
...
+      require 'releaser_json.php';
+      print_current_stable_version('$json');
+    ")
+
+    if [[ "$major_version" != "$current_stable_version" ]]; then
+      prompt new_status_for_previous_stable "Since this release is stable, what should be the new status for the current stable version ($current_stable_version)?" testing lts eol
+
+      json=$(php -r "
+        require 'releaser_json.php';
+        update_version_status('$json', '4.2', '$new_status_for_previous_stable');
```
Comment (colemanw) [inline tools/scripts/releaser/releaser]: looks like `4.2` was meant to be a variable?

File: tools/scripts/releaser/releaser_json.php [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+ * @param string $major_version A major version string, e.g. 1.1
+ * @param string $release_properties_json JSON string containing properties
+ *  for the new release, as seen in latest.civicrm.org/versions.json
+ *  {$major_version:{'releases':[$release_properties_json]}}
+ */
+function add_release($json, $major_version, $release_properties_json) {
+  $versions = json_decode($json, TRUE);
+  $release_properties = json_decode($release_properties_json, TRUE);
+  if (array_key_exists('security', $release_properties)) {
+    mylog($release_properties, '$release_properties');
```
Comment (colemanw) [inline tools/scripts/releaser/releaser_json.php]: stray debug statement?

**Comment by twomice** [comment]
OK, removed that debugging code and hardcoded version string. Thanks!

**Comment by twomice** [comment]
Please hold on this. I need to add support for enforcing "only one lts at a time", /and/ fix a bug with updating the status on the new release.

**Comment by twomice** [comment]
With the above commit, the if/then nesting gets a little hairy, aiming to handle status changes for older versions. The pseudocode is something like this:

    ask for $status
    if $status is lts
      if $older_lts_version exists
        ask $new_status for $older_lts_version
        mark $new_status for $older_lts_version
    elif $status is stable
      if $older_stable_version exists
        ask $new_status for $older_stable_version
        mark $new_status for $older_stable_version
        if $new_status == 'lts'
          if $older_lts_version exists
            ask $new_status for $older_lts_version
            mark $new_status for $older_lts_version

**Comment by colemanw** [comment]
Code looks good to me.
@twomice if you've run this through a few paces then I'll go ahead and merge it.

**Comment by twomice** [comment]
Yes, I've run it through all I could think of.

**Comment by colemanw** [comment]
@kurund you'll need to run the `self-update` script to pull this new code to the releaser script on sushi.

**Pull request merged**